src/maplibre_helpers.ts

```diff
@@ -86,20 +84,39 @@ export function overwriteSource(
   map.addSource(id, source);
 }
 
-export function overwriteLayer(
-  map: Map,
-  layer,
-  beforeId: string | undefined = undefined
-) {
+// The layer.id here MUST be present in layerZorder.
+export function overwriteLayer(map: Map, layer) {
   if (map.getLayer(layer.id)) {
```

Comment:
Starting a slightly orthogonal conversation here: using these two `overwrite` methods is kind of gross; see https://github.com/acteng/atip#hmr-best-practices for why. I see people using Svelte components to specify GeoJSON sources and layers in https://svelte-maplibre.vercel.app/examples/geojson_polygon, https://www.npmjs.com/package/@mapcomponents/react-maplibre, etc. https://imfeld.dev/writing/svelte_domless_components and other articles on the same blog explain the reasoning here. I think we should consider switching to this style too.

Also, if you edit this file and let Vite auto-reload, you'll see it crash! You have to hard refresh. It's because while some component gets recreated the maplibre `Map` is null, and we try to `addSource`. We need to figure out how to robustly add stuff to the map, or defer it to when the map is loaded, and handle all the weird Svelte component orders